src/main/scala/com/github/sashjakk/App.scala

```diff
@@ -1,48 +1,57 @@
 package com.github.sashjakk
 
+import cats.data.EitherT
+import cats.effect.{IO, IOApp}
 import com.github.sashjakk.spot.book.{BookingCreate, BookingRepo}
 import com.github.sashjakk.spot.share.{ShareCreate, SpotShareRepo}
 import com.github.sashjakk.spot.{SpotCreate, SpotRepo, SpotService}
-import com.github.sashjakk.user.{UserCreate, UserRepo, UserService}
+import com.github.sashjakk.user.{DuplicatePhoneNumberError, PersistenceError, User, UserCreate, UserRepo, UserService}
 
 import java.time.{LocalDateTime, ZoneOffset}
 
-object App {
-  def main(args: Array[String]): Unit = {
-    val userRepo = UserRepo.inMemory()
-    val userService = UserService.default(userRepo)
+object App extends IOApp.Simple {
+  val run: IO[Unit] = {
+    val userRepo = UserRepo.inMemory[IO]()
+    val userService = UserService.make(userRepo)
 
-    val spotRepo = SpotRepo.inMemory()
-    val spotShareRepo = SpotShareRepo.inMemory()
-    val bookSpotRepo = BookingRepo.inMemory()
-    val spotService = SpotService.default(userRepo, spotRepo, spotShareRepo, bookSpotRepo)
+    val spotRepo = SpotRepo.inMemory[IO]()
+    val spotShareRepo = SpotShareRepo.inMemory[IO]()
+    val bookSpotRepo = BookingRepo.inMemory[IO]()
+    val spotService = SpotService.make[IO](userRepo, spotRepo, spotShareRepo, bookSpotRepo)
 
     val user1 = UserCreate(name = "John", surname = "Smith", phone = "+37127231766")
     val user2 = UserCreate(name = "Jenny", surname = "Penny", phone = "+37166713273")
 
     val now = LocalDateTime.now()
 
-    val result = for {
-      user <- userService.create(user1)
-      user2 <- userService.create(user2)
+    for {
+      user <- EitherT(userService.create(user1)).leftMap {
+        case DuplicatePhoneNumberError => new Error("duplicate phone number")
+        case PersistenceError          => new Error("failed to save")
+      }.rethrowT
```

Comment:
😅 looks strange... would be much easier to IO.raiseError directly without remapping

Reply:
You don't have to use either, esecially if you have exceptions in Left.